qos-service-impl/src/main/java/com/palantir/atlasdb/qos/ratelimit/QosRateLimiter.java

```diff
@@ -95,8 +98,21 @@ public Duration consumeWithBackoff(long estimatedNumUnits) {
      * overhead and double comparisons, we maintain the current rate ourselves.
      */
     private void updateRateIfNeeded() {
-        if (currentRate != unitsPerSecond.get()) {
-            createRateLimiterAtomically();
+        if (rateUpdateLimiter.tryAcquire()) {
+            long updatedRate = getUpdatedRate();
```

Comment:
It might still be nice to not add any network overhead to a request, even if it's time to fetch a new limit. Could we just have a background scheduled executor that fetches new limits, then updates the rate limiter when there's a change?